pkg/wsman/hash.go

```diff
@@ -124,6 +124,40 @@ func (c *authChallenge) authorize(method, uri string) (string, error) {
 	return sb.String(), nil
 }
 
+func matchesDelim(r rune, delims []rune) bool {
+	for _, d := range delims {
+		if r == d {
+			return true
+		}
+	}
+	return false
+}
+
+func splitQuotedFields(input string, delimeters string) []string {
+	singleQuote := false
+	doubleQuote := false
+	var delimRunes = []rune(delimeters)
+	var out []string
+
+	sb := &strings.Builder{}
+	for _, r := range input {
+		if r == '"' && !singleQuote {
+			doubleQuote = !doubleQuote
+		} else if r == '\'' && !doubleQuote {
+			singleQuote = !singleQuote
+		} else if !singleQuote && !doubleQuote && matchesDelim(r, delimRunes) {
+			out = append(out, sb.String())
+			sb.Reset()
+		} else {
+			sb.WriteRune(r)
+		}
+	}
+	if sb.Len() > 0 {
+		out = append(out, sb.String())
+	}
+	return out
+}
```

Comment:
can be simplified to match how RPS handles this:
``` suggestion
func splitQuotedFields(input string) []string {
	r := regexp.MustCompile(`([^=,]*)=("[^"]*"|[^,"]*)`)
	return r.FindAllString(input, -1)
}
```

Reply:
OK, ammended the commit.